epregressions/diffs/ci_compare_script.py

```diff
@@ -257,55 +257,40 @@ def main_function(file_name, base_dir, mod_dir, base_sha, mod_sha, make_public,
                 # print("Processing output file: {0}, uploading to: {1}".format(filepath_to_send, filepath))
 
                 key = boto.s3.key.Key(bucket, file_path)
-                file_to_send = open(file_path_to_send, 'r')
-                key.set_contents_from_string(file_to_send.read())
-
-                if make_public:
-                    key.make_public()
-
-                htmlkey = boto.s3.key.Key(bucket, file_path + ".html")
-                htmlkey.set_contents_from_string("""
+                with open(file_path_to_send, 'r') as file_to_send:
```

Comment:
Using `with` here makes sure to close the file when exiting the scope.  I'm not sure if this was causing any problems, but this should make sure it doesn't cause any.